src/Symfony/Component/Serializer/Tests/Fixtures/StaticConstructorInstantiator.php

```diff
@@ -11,22 +11,24 @@
 
 namespace Symfony\Component\Serializer\Tests\Fixtures;
 
-use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Instantiator\Instantiator;
 
 /**
- * @author Guilhem N. <egetick@gmail.com>
+ * @author Baptite Leduc <baptiste.leduc@gmail.com>
  */
-class StaticConstructorNormalizer extends ObjectNormalizer
+class StaticConstructorInstantiator extends Instantiator
```

Comment:
Couldn't we use composition instead?

Reply:
I don't see what you would prefer here, could you share me a example ?

Reply:
For this case, it's not possible since I need to replace the behavior of one of the Instantiator's methods

Reply:
Let's try to find a way to to support static constructors without extending `Instantiator`